bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/DaikinHandlerFactory.java

```diff
@@ -31,22 +31,33 @@
 
  */
 @Component(service = ThingHandlerFactory.class, configurationPid = "binding.daikin")
-@NonNullByDefault
```

Comment:
This was not what I meant. Please revert that and create a constructor like that

```
private final DaikinDynamicStateDescriptionProvider stateDescriptionProvider;

@Activate
public DaikinHandlerFactory(@Reference DaikinDynamicStateDescriptionProvider stateDescriptionProvider) {
  this.stateDescriptionProvider = stateDescriptionProvider;
}
```

and remove the set/unset methods.

Reply:
Thanks for this pointer!

Since it will be final, these functions won't work. Should I remove them?
```

    @Reference
    protected void setDynamicStateDescriptionProvider(DaikinDynamicStateDescriptionProvider stateDescriptionProvider) {
        this.stateDescriptionProvider = stateDescriptionProvider;
    }

    protected void unsetDynamicStateDescriptionProvider(DaikinDynamicStateDescriptionProvider stateDescriptionProvider) {
        this.stateDescriptionProvider = null;
    }
```